source/microbit/microbitimage.cpp

```diff
@@ -439,6 +381,39 @@ mp_obj_t microbit_image_fill(mp_obj_t self_in, mp_obj_t n_in) {
 }
 MP_DEFINE_CONST_FUN_OBJ_2(microbit_image_fill_obj, microbit_image_fill);
 
+mp_obj_t microbit_image_blit(mp_uint_t n_args, const mp_obj_t *args) {
+    microbit_image_obj_t *self = (microbit_image_obj_t*)args[0];
+    check_mutability(self);
+
+    mp_obj_t src = args[1];
+    if (mp_obj_get_type(src) != &microbit_image_type) {
+        nlr_raise(mp_obj_new_exception_msg(&mp_type_TypeError, "expecting an image"));
+    }
+    if (n_args == 7) {
+        nlr_raise(mp_obj_new_exception_msg(&mp_type_TypeError,
+            "must specify both offsets"));
+    }
+    mp_int_t x = mp_obj_get_int(args[2]);
+    mp_int_t y = mp_obj_get_int(args[3]);
+    mp_int_t w = mp_obj_get_int(args[4]);
+    mp_int_t h = mp_obj_get_int(args[5]);
+    if (w < 0 || h < 0) {
+        nlr_raise(mp_obj_new_exception_msg(&mp_type_ValueError,
+            "size cannot be negative"));
+    }
```

Comment:
Should we have a check that we are not blitting onto itself? (Or, better, if we are, that the areas don't overlap.) The results of that could be funky...